### PR TITLE
authenticate: add jwks and .well-known endpoint

### LINKED ISSUE #189: Expose (public) Signing Key for JWT
**Is your feature request related to a problem? Please describe.**

Hey Pomerium devs,

I'm deploying Istio (https://istio.io/) with Pomerium - I'd like to use the security features of Isto ( namely, checking JWT assertions for specific routes - https://istio.io/docs/reference/config/istio.authentication.v1alpha1/)

At the moment, I have a separate service which simply serves the public signing key, so i can check JWT assertions using the jwks_uri parameter in Istio.

**Describe the solution you'd like**

Implement the jwksuri as per openid specification (https://openid.net/specs/openid-connect-discovery-1_0.html, jwks_uri) - Or just expose a route like auth0 does (https://auth0.com/docs/jwks - their example is https://YOUR_DOMAIN/.well-known/jwks.json).

**Describe alternatives you've considered**

Handing keys out via an out of band mechanism is cumbersome - in my case, if it wasn't for an internal web service it would end up being a manual copy-paste into kubernetes manifest files for other services. 

I had a quick look through the documentation (which didn't seem to mention anything) and had a look at the routes exposed within the project and didn't see anything like this currently - Apologies if it already exists!

Thanks / Go beyondcorp!

### PR DESCRIPTION
**Pull request by desimone**
# Summary 

Screen Shot 2020-05-20 at 9 09 03 PM

Screen Shot 2020-05-20 at 9 08 54 PM

## Related issues
- Fixes #189 

### See also

- https://www.googleapis.com/oauth2/v3/certs
- https://tools.ietf.org/html/rfc7517 
- https://auth0.com/docs/tokens/concepts/jwks
- https://openid.net/specs/openid-connect-discovery-1_0.html

**Checklist**:

- [x] add related issues
- [ ] updated docs (TODO!)
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

File: internal/controlplane/xds_routes.go [hunk 1]
```diff
@@ -45,6 +45,8 @@ func (srv *Server) buildPomeriumHTTPRoutes(options *config.Options, domain strin
 		srv.buildControlPlanePathRoute("/healthz"),
 		srv.buildControlPlanePathRoute("/.pomerium"),
 		srv.buildControlPlanePrefixRoute("/.pomerium/"),
+		srv.buildControlPlanePathRoute("/.well-known"),
+		srv.buildControlPlanePrefixRoute("/.well-known/"),
```
Comment (calebdoxsey) [inline internal/controlplane/xds_routes.go:49]: should we limit this to `/.well-known/pomerium/`?

Comment (desimone) [reply internal/controlplane/xds_routes.go]: Yes, good idea.

File: authenticate/handlers.go [hunk 2]
```diff
@@ -72,12 +73,55 @@ func (a *Authenticate) Mount(r *mux.Router) {
...
 	api.Path("/v1/refresh").Handler(httputil.HandlerFunc(a.RefreshAPI))
 }
 
+// Well-Known Uniform Resource Identifiers (URIs)
+// https://en.wikipedia.org/wiki/List_of_/.well-known/_services_offered_by_webservers
+func (a *Authenticate) wellKnown(w http.ResponseWriter, r *http.Request) error {
+	wellKnownURLS := struct {
+		// URL string referencing the client's JSON Web Key (JWK) Set
+		// RFC7517 document, which contains the client's public keys.
+		JSONWebWeySetURL       string `json:"jwks_uri"`
```
Comment (calebdoxsey) [inline authenticate/handlers.go:94]: typo: `JSONWebWeySetURL` should be `JSONWebKeySetURL`

File: docs/configuration/readme.md [hunk 3]
```diff
@@ -1034,9 +1033,31 @@ See [ProxyPreserveHost](http://httpd.apache.org/docs/2.0/mod/mod_proxy.html#prox
...
+      "alg": "ES256",
+      "x": "QCN7adG2AmIK3UdHJvVJkldsUc6XeBRz83Z4rXX8Va4",
+      "y": "PI95b-ary66nrvA55TpaiWADq8b3O1CYIbvjqIHpXCY"
+    }
+  ]
+}
+```
 
-If no certificate is specified, one will be generated for you and the base64'd public key will be added to the logs.
+If no certificate is specified, one will be generated and the base64'd public key will be added to the logs. Note, hoever, that this key be unique to each service, ephemerial, and will not be accessible via the authenticate service's `jwks_uri` endpoint.
```
Comment (calebdoxsey) [inline docs/configuration/readme.md:1060]: typo: `hoever` should be `however`, `ephemerial` should be `ephemeral`

Comment (desimone) [reply docs/configuration/readme.md:1060]: @calebdoxsey I'm having one of those days...

**Pull request merged**